index.js

```diff
@@ -28,130 +15,137 @@ const colorCodes = {
+        ? options
+        : options && options.transporter
+        ? options.transporter
+        : undefined;
+
+    // eslint-disable-next-line func-names
+    return function printFunc(...args) {
+      const string = util.format(...args);
+      if (transporter) transporter(string, args);
+      else console.log(...args);
+    };
+  })();
+
+  // eslint-disable-next-line func-names
+  return async function logger(ctx, next) {
     // request
-    const start = ctx[Symbol.for('request-received.startTime')] ? ctx[Symbol.for('request-received.startTime')].getTime() : Date.now()
-    print('  ' + chalk.gray('<--') +
-      ' ' + chalk.bold('%s') +
-      ' ' + chalk.gray('%s'),
-    ctx.method,
-    ctx.originalUrl)
+    const start = ctx[Symbol.for('request-received.startTime')]
+      ? ctx[Symbol.for('request-received.startTime')].getTime()
+      : Date.now();
+    print(
+      '  ' +
+        chalk.gray('<--') +
+        ' ' +
+        chalk.bold('%s') +
+        ' ' +
+        chalk.gray('%s'),
+      ctx.method,
+      ctx.originalUrl
+    );
 
     try {
-      await next()
+      await next();
     } catch (err) {
       // log uncaught downstream errors
-      log(print, ctx, start, null, err)
-      throw err
+      log(print, ctx, start, null, err);
+      throw err;
     }
 
     // calculate the length of a streaming response
     // by intercepting the stream with a counter.
     // only necessary if a content-length header is currently not set.
-    const length = ctx.response.length
-    const body = ctx.body
-    let counter
-    if (length == null && body && body.readable) {
-      ctx.body = body
-        .pipe(counter = Counter())
-        .on('error', ctx.onerror)
-    }
+    const {
+      body,
+      response: { length }
+    } = ctx;
+    let counter;
+    if (length === null && body && body.readable)
+      ctx.body = body.pipe((counter = counterFunc())).on('error', ctx.onerror);
 
     // log when the response is finished or closed,
     // whichever happens first.
-    const res = ctx.res
+    const { res } = ctx;
 
-    const onfinish = done.bind(null, 'finish')
-    const onclose = done.bind(null, 'close')
+    const onfinish = done.bind(null, 'finish');
+    const onclose = done.bind(null, 'close');
 
-    res.once('finish', onfinish)
-    res.once('close', onclose)
+    res.once('finish', onfinish);
+    res.once('close', onclose);
 
-    function done (event) {
-      res.removeListener('finish', onfinish)
-      res.removeListener('close', onclose)
-      log(print, ctx, start, counter ? counter.length : length, null, event)
+    function done(event) {
+      res.removeListener('finish', onfinish);
+      res.removeListener('close', onclose);
+      log(print, ctx, start, counter ? counter.length : length, null, event);
     }
-  }
-}
-
-/**
- * Log helper.
- */
+  };
+};
 
-function log (print, ctx, start, len, err, event) {
+// Log helper.
+// eslint-disable-next-line max-params
```

Comment:
I don't like the lint rule, it's too restrictive, and change the code style is useless to this module.

Reply:
I will add some rules to XO config ...